Utilities/bootstrap

```diff
@@ -886,7 +921,37 @@ def build_llbuild(args):
     cmake_cache_path = os.path.join(llbuild_build_dir, "CMakeCache.txt")
     if not os.path.isfile(cmake_cache_path) or not args.swiftc_path in open(cmake_cache_path).read():
         mkdir_p(llbuild_build_dir)
-        cmd = ["cmake", "-G", "Ninja", "-DCMAKE_BUILD_TYPE:=Debug", "-DCMAKE_C_COMPILER:=clang", "-DCMAKE_CXX_COMPILER:=clang++", "-DLLBUILD_SUPPORT_BINDINGS:=Swift", "-DSWIFTC_EXECUTABLE:=%s" % (args.swiftc_path), llbuild_source_dir]
+
+        if platform.system() == 'Darwin' or platform.system() == 'Linux':
+            # NOTE(compnerd) ideally, we should use the Unix command 'cc' which
+            # would allow users to use the C compiler of their choice
+            CC = os.environ.get('CC', 'clang')
+            # NOTE(compnerd) ideally, we would use the Unix command 'c++' which
+            # would allow users to use the C++ compiler of their choice
+            CXX = os.environ.get('CXX', 'clang++')
```

Comment:
@compnerd Want to just propose these changes in a separate PR?

Reply:
Sure, I can split this out into a separate change.

Reply:
Please avoid `if Darwin || Linux else Windows` style constructs and prefer `if Windows || Darwin else`... the else case should be able to handle the majority case without having to list every Unix OS in the world for which the Darwin/Linux case works perfectly fine.